react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+
+  useEffect(() => {
+    async function work() {
+      try {
+        const newItems = await readItems();
+        setItems(newItems);
+        setIsLoading(false);
+      } catch (error) {
+        setError(error);
+        setIsLoading(false);
```

Comment:
You probably want to log the error as well. And putting `setIsLoading(false)` into a `finally` makes the code a tiny bit cleaner.